src/SFA.DAS.AssessorService.Application.Api/Orchestrators/OrganisationSearchOrchestrator.cs

```diff
@@ -0,0 +1,410 @@
+                    if (response != null) results.AddRange(response);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError($"Error from ATP Register. {name} , Message: {ex.Message}");
+                }
+            }
+
+            if (exactNames != null)
+            {
+                foreach (var exactName in exactNames)
+                {
+                    try
+                    {
+                        var response = await _roatpApiClient.SearchOrganisationByName(exactName, true);
+                        if (response != null) results.AddRange(response);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError($"Error from ATP Register. Exact Name: {exactName} , Message: {ex.Message}");
+                    }
+                }
+            }
+
+            return results.GroupBy(r => r.Ukprn).Select(group => group.First()).ToList();
+        }
+
+        private async Task<IEnumerable<OrganisationSearchResult>> GetProviderRegisterResults(string name, IEnumerable<string> exactNames, int? ukprn)
+        {
+            var results = new List<OrganisationSearchResult>();
+
+            if (ukprn.HasValue)
+            {
+                try
+                {
+                    var ukprnResponse = await _roatpApiClient.GetOrganisationByUkprn(ukprn.Value);
+                    if (ukprnResponse != null) results.Add(ukprnResponse);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError($"Error from Provider Register. UKPRN: {ukprn.Value} , Message: {ex.Message}");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(name))
+            {
+                try
+                {
+                    var response = await _roatpApiClient.SearchOrganisationByName(name, false);
+                    if (response != null) results.AddRange(response);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError($"Error from Provider Register. {name} , Message: {ex.Message}");
+                }
+            }
+
+            if (exactNames != null)
+            {
+                foreach (var exactName in exactNames)
+                {
+                    try
+                    {
+                        var response = await _roatpApiClient.SearchOrganisationByName(exactName, true);
+                        if (response != null) results.AddRange(response);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError($"Error from Provider Register. Exact Name: {exactName} , Message: {ex.Message}");
+                    }
+                }
+            }
+
+            return results.GroupBy(r => r.Ukprn).Select(group => group.First()).ToList();
+        }
```

Comment:
The orchestrator should not have to worry about all this. If this is decoupled from the orchestrator and moved to a `RegisterResults` class, the orchestrator can simply orchestrate by asking that class to get the type of results it needs. This also means that the logic in the new `RegisterResults` class can be unit tested independently of the orchestrator.